nilearn/image/image.py

```diff
@@ -902,12 +908,22 @@ def clean_img(imgs, sessions=None, detrend=True, standardize=True,
                 "specified None. imgs header suggest it to be {0}".format(
                     imgs.header.get_zooms()[3]))
 
-    data = signal.clean(
-        imgs_.get_data().reshape(-1, imgs_.shape[-1]).T, sessions=sessions,
-        detrend=detrend, standardize=standardize, confounds=confounds,
-        low_pass=low_pass, high_pass=high_pass, t_r=t_r,
-        ensure_finite=ensure_finite).T.reshape(imgs_.shape)
-    return new_img_like(imgs, data, copy_header=True)
+    # Store function parameters in dictionary for easy access
```

Comment:
I think it would make easier if we would stick to mask_img implementation rather than cleaning the API for easy access.

Reply:
Not sure what you mean by that. Having a mask as default? One could imagine `mask=np.ones(imgs_.shape[:3])`. But the missing header issue from `unmask` would persist. But could be solved with a `get/set_zooms`.

Reply:
Nope keeping code as it was and just modifying the first positional argument. For me, only thing which varies is first argument to signal.clean

Reply:
I thought about that. So we would have two possible `signals` as first argument? But this would mean to have two `if`s, which I wanted to prevent.
```python
if mask_img is not None:
    signals = masking.apply_mask(imgs_, mask_img_)
else:
    signals = imgs_.get_data().reshape(-1, imgs_.shape[-1]).T
data = signal.clean(signals,...)
if mask_img is not None:
    data = masking.unmask(sigs_clean, mask_img_).get_data()
return new_img_like(imgs_, data, copy_header=True)
```